File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
        public int findMin(int[] nums) {
            int leftPointer = 0;
            int rightPointer = nums.length-1;

            while (leftPointer < rightPointer){
                int midPoint = leftPointer + Math.floorDiv((rightPointer - leftPointer), 2);
                if(nums[midPoint] > nums[rightPointer]){
                    leftPointer = midPoint + 1;
                }else{
                    rightPointer = midPoint;
                }
            }
            return nums[leftPointer];

    }

}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        min_value = nums[0]
        left_pointer = 0
        right_pointer = len(nums) - 1

        while left_pointer < right_pointer:
            mid_point = left_pointer + (right_pointer - left_pointer) // 2
            if  nums[mid_point] > nums[right_pointer]:
                left_pointer = mid_point + 1
            else:
                right_pointer = mid_point
        return nums[left_pointer]